### PR TITLE
Upgrade Cypress to 7.5.0

### PR DESCRIPTION
**Pull request by ddzz**
## Description
This PR upgrades Cypress to `7.5.0`, which is the latest version.

## Testing done
- [ ] Tests pass on CI.

## Acceptance criteria
- [ ] Tests pass on CI.

### CODE REVIEW

File: src/applications/financial-status-report/tests/e2e/fsr-5655-wizard.cypress.spec.js
Comment (Mottie) [inline src/applications/financial-status-report/tests/e2e/fsr-5655-wizard.cypress.spec.js:16]: I'm confused by this update. Why do some `intercept` URLs include a `**` prefix and this one doesn't? Also why isn't https://github.com/department-of-veterans-affairs/vets-website/pull/17131/files#diff-52b2fcb0444f109e051568e09ffbe1e77efae3baeb75371590f17e7e9064731bR73 getting modified at all??

File: src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js
Comment (billfienberg) [inline src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js:14]: Dumb question: What is `messaging` in this context, and why does it need to be changed from `true` to `false`?

**Review by billfienberg — APPROVED** [review]
LGTM

**Review by 1Copenut — APPROVED** [review]
LGTM!

File: src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js
Comment (ddzz) [reply src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js:14]: This test started failing due to the Cypress upgrade, and it may be a bug in the UI itself and not just an issue with the test. An `undefined` appears in the UI when running this test with the latest version of Cypress. `messaging` is used in the UI code to render different components. I'm not familiar with this feature to determine why this fixed the failure, so I'm going to get a review from the team that owns this test before I merge.

**Review by timwright12 — APPROVED** [review]

File: src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js
Comment (erikphansen) [reply src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js:14]: I'll look into this.

**Review by erikphansen — APPROVED** [review]

**Review by micahchiang — APPROVED** [review]

**Review by craigwheeler — APPROVED** [review]

**Review by Mottie — APPROVED** [review]

File: src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js
Comment (ddzz) [reply src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js]: @erikphansen is it fine if I disable this test until you revisit it? This PR may have exposed a UI bug here. I'm seeing an `undefined` in the UI:
Screen Shot 2021-05-21 at 8 40 43 AM

Comment (erikphansen) [reply src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js]: yes, feel free to disable it

File: src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js
Comment (erikphansen) [inline src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js:21]: @ddzz I figured out the issue. In Cypress 6.9.1, this intercept worked for both the `GET /v0/messaging/health/folders/0` and also the `GET /v0/messaging/health/folders/0/messages` endpoints. Both are used by the component under test.

But in Cypress 7.3, this intercept _only_ mocks the `GET /v0/messaging/health/folders/0` endpoint.

Also note that this mocked error won't really be treated as an error since we didn't set the status code. This mock should look like this:
```
cy.intercept('GET', '/v0/messaging/health/folders/0', {
    statusCode: 400,
    body: ERROR_400,
});
```

So, what happens in Cypress 7.3 is the call to `/v0/messaging/health/folders/0/messages` fails with a 4XX because it isn't mocked at all. And, sure enough, our error handling code is not correct. So, thanks Cypress for revealing a real issue. I'm opening a PR to fix our error handling code. Feel free to leave this test skipped.

Comment (ddzz) [reply src/applications/personalization/dashboard-2/tests/e2e/messaging-error.cypress.spec.js]: Thanks for looking into this! I suspected it was a bug, thanks for confirming.

File: package.json
Comment (pjhill) [inline package.json:160]: Was this supposed to be "^7.5.0"?

Comment (ddzz) [reply package.json:160]: Yeah, I had a typo when I ran the `yarn upgrade`. Opened a PR to fix this: https://github.com/department-of-veterans-affairs/vets-website/pull/17556

**Pull request merged**
